backend/lib/endpoints/users.js

```diff
@@ -6,39 +6,31 @@ const httpErrors = require("http-errors");
 async function routes(app) {
   const User = app.mongo.model("User");
 
-  app.get(
-    "/current",
-    { preValidation: [app.authenticate] },
-    async (req, reply) => {
-      // todo: get current user from JWT
-      const result = await User.findById("");
-      if (result === null) {
-        return reply.send(new httpErrors.NotFound());
-      }
-      return {
-        id: result._id,
-        firstName: result.firstName,
-        lastName: result.firstName,
-        email: result.email,
-      };
-    },
-  );
+  app.get("/current", { preValidation: [app.authenticate] }, async () => {
+    // todo: get current user from JWT
+    const result = await User.findById("");
+    if (result === null) {
+      return new httpErrors.NotFound();
+    }
+    return {
```

Comment:
Not too familiar with Mongoose, but is there a reason for not returning the `result` object as is? Does it contain extraneous properties that we would not wish to return as part of the response?

Reply:
Yes, I just changed thsi since I read in the Fastify documentation that it's recommended, it has nothing to do with Mongoose. 
It's a convention for async routes in Fastify handlers and it's done this way because of how JavaScript workes with Promises and async/await, you can read more in the docs: https://www.fastify.io/docs/latest/Routes/#promise-resolution

Reply:
And ah maybe I misunderstood your question, I was still thinking about some of the responses, sometimes it might not be the best to use a NotFound error but we return an empty list, depending on the request.
For the `result` we definitely need to be careful because we might expose data that we don't want, in this case `/current` only requests your own  logged in user data so I guess we can probably return the full user object.

Reply:
Makes sense, I was just curious about returning only a few fields in the result rather than all the fields, but from a security standpoint it does make sense to whitelist a few fields that we want to return as part of the response.